src/leafletControl.js

```diff
@@ -220,9 +221,15 @@ const Control = {
   },
 
   addMarker(result) {
-    const { marker: options, showPopup } = this.options;
+    const { marker: options, showPopup, popupFormat } = this.options;
     const marker = new L.Marker([result.y, result.x], options);
-    marker.bindPopup(result.label);
+    let popupLabel = result.label;
+
+    if (typeof popupFormat === 'function') {
+      popupLabel = popupFormat(result);
```

Comment:
Change `popupFormat` signature to `popupFormat({ query, result })`. This does require the `addMarker` method to be called with  addMarker(result, query)`, as well as a similar change to `showResult`, but gives the end user ultimate flexibility in styling his popup.